core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/FastPassGrantComponent.scala

```diff
@@ -0,0 +1,153 @@
+package org.broadinstitute.dsde.rawls.dataaccess.slick
+
+import org.broadinstitute.dsde.rawls.model._
+import org.broadinstitute.dsde.workbench.model.google.iam.{IamMemberTypes, IamResourceTypes}
+import org.joda.time.DateTime
+
+import java.sql.Timestamp
+import java.util.UUID
+import scala.language.postfixOps
+
+/**
+ * Created by tlangs on 3/16/2023.
+ */
+case class FastPassGrantRecord(
+  id: Long,
+  workspaceId: UUID,
+  userSubjectId: String,
+  accountEmail: String,
+  accountType: String,
+  resourceType: String,
+  resourceName: String,
+  roleName: String,
+  expiration: Timestamp,
+  created: Timestamp
+)
+
+object FastPassGrantRecord {
+  def fromFastPassGrant(fastPassGrant: FastPassGrant): FastPassGrantRecord =
+    FastPassGrantRecord(
+      fastPassGrant.id,
+      UUID.fromString(fastPassGrant.workspaceId),
+      fastPassGrant.userSubjectId.value,
+      fastPassGrant.accountEmail.value,
+      fastPassGrant.accountType.value,
+      fastPassGrant.resourceType.value,
+      fastPassGrant.resourceName,
+      fastPassGrant.organizationRole,
+      new Timestamp(fastPassGrant.expiration.getMillis),
+      new Timestamp(fastPassGrant.created.getMillis)
+    )
+
+  def toFastPassGrant(fastPassGrantRecord: FastPassGrantRecord): FastPassGrant =
+    FastPassGrant(
+      fastPassGrantRecord.id,
+      fastPassGrantRecord.workspaceId.toString,
+      RawlsUserSubjectId(fastPassGrantRecord.userSubjectId),
+      RawlsUserEmail(fastPassGrantRecord.accountEmail),
+      IamMemberTypes.withName(fastPassGrantRecord.accountType),
+      IamResourceTypes.withName(fastPassGrantRecord.resourceType),
+      fastPassGrantRecord.resourceName,
+      fastPassGrantRecord.roleName,
+      new DateTime(fastPassGrantRecord.expiration),
+      new DateTime(fastPassGrantRecord.created)
+    )
+}
+
+trait FastPassGrantComponent {
+  this: DriverComponent with WorkspaceComponent =>
+
+  import driver.api._
+
+  class FastPassGrantTable(tag: Tag) extends Table[FastPassGrantRecord](tag, "FASTPASS_GRANTS") {
+
+    def id = column[Long]("id", O.PrimaryKey, O.AutoInc)
+
+    def workspaceId = column[UUID]("workspace_id")
+
+    def userSubjectId = column[String]("user_subject_id", O.PrimaryKey)
```

Comment:
`userSubjectId` is the unique ID used by Sam for a user in Terra. Its used here to tie all the Pet Service Accounts of a user together so that they're removed when a user is removed.